Documentation/network/pod-mac-address.rst

```diff
@@ -0,0 +1,69 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _pod_mac_address:
+
+************************************
+Use a Specific MAC Address for a Pod
+************************************
+
+
+Some applications bind software licenses to network interface MAC addresses.
+Typically, the solution is very complex and requires things like Docker within
+the cluster.
+
+Therefore, cilium provides the ability to specific MAC addresses
+for pods instead of letting the operating system allocate them.
+
+
+
+
+Specific MAC Address
+#####################
+
+The MAC address configured by the annotation described here will be visible
+from within the container on the eth0 interface. Since it is isolated to the
+container it will not collide with any other MAC addresses assigned to other
+pods on the same node.
+
+
+Annotate the pod with ``cni.cilium.io/mac-address`` set to the desired MAC address.
+After adding the annotation to the pod, it appears like so.
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      annotations:
+        cni.cilium.io/mac-address: e2:9c:30:38:52:61
+      labels:
+        app: busybox
+      name: busybox
+      namespace: default
+
+After the pod runs successfully, we can query the pod's eth0 interface.
+Check whether its mac address is the specified mac address.
+
+.. code-block:: shell-session
+
+    $ kubectl exec -it busybox -- ip addr
+    1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
+        link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+        inet 127.0.0.1/8 scope host lo
+           valid_lft forever preferred_lft forever
+        inet6 ::1/128 scope host
+           valid_lft forever preferred_lft forever
+    7: eth0@if8: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1500 qdisc noqueue qlen 1000
+        link/ether e2:9c:30:38:52:61 brd ff:ff:ff:ff:ff:ff
+        inet 10.244.2.195/32 scope global eth0
+           valid_lft forever preferred_lft forever
+        inet6 fe80::e46d:f4ff:fe4d:ebca/64 scope link
+           valid_lft forever preferred_lft forever
+
+.. warning::
+    The MAC address must be specified in this annotation during Pod creation,
+    and Cilium will not respect changes to the annotation after Pod creation.
```

Comment:
I think that this is critical information, and should be worked into the text directly rather than adding a warning callout at the end of the document.